### PR TITLE
Binding to a project not part to our organization results in NPR

### LINKED ISSUE #382: Binding to a project not part to our organization results in NPR
Trying to bind to a project not part of our organization on SonarCloud will result in an error:
`SonarQube request failed: Object reference not set to an instance of an object. Cannot download the quality profile for language C#`

### LINKED ISSUE #387: Unexpected character encountered error
When trying to connect to the following URL (wrong website): https://about.sonarcloud.io/
SonarLint for VS will fail with the following message: `SonarQube request failed: Unexpected character encountered while parsing value: <. Path '', line 0, position 0. `

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #382 and fix #387

### CODE REVIEW

File: src/SonarQube.Client/Services/SonarQubeClient.cs [hunk 1]
```diff
@@ -267,6 +290,16 @@ private static IEnumerable<T> ReadFromProtobufStream<T>(Stream stream, MessagePa
             return new Result<T>(httpResponse, parseStringResult(stringResponse));
         }
 
+        private void ThrowsIfError(JObject parsedJson)
```
Comment (Evangelink) [inline src/SonarQube.Client/Services/SonarQubeClient.cs]: We should probably came up with a better solution but the kind of return depends on the API and the SQ version so I went for the easy path.

**Comment by Evangelink** [comment]
This change will display a more helpful message than the NPR but won't solve the issue. Issue was solved by @valhristov last PR.

File: src/SonarQube.Client/Services/SonarQubeClient.cs [hunk 2]
```diff
@@ -267,6 +290,16 @@ private static IEnumerable<T> ReadFromProtobufStream<T>(Stream stream, MessagePa
...
+        {
+            var firstErrorMessage = parsedJson["errors"]?.Select(x => x["msg"]?.ToObject<string>())
+                .FirstOrDefault();
+            if (firstErrorMessage != null)
+            {
+                throw new Exception(firstErrorMessage);
```
Comment (SonarTech) [inline src/SonarQube.Client/Services/SonarQubeClient.cs]: MAJOR 'System.Exception' should not be thrown by user code. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS112)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/SonarQube.Client/Services/SonarQubeClient.cs (hunk 1 above)
Comment (duncanp-sonar) [reply src/SonarQube.Client/Services/SonarQubeClient.cs]: How difficult would be it be to add a unit test, at least for the API that was causing the failure?

File: src/SonarQube.Client/Services/SonarQubeClient.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/SonarQube.Client/Services/SonarQubeClient.cs]: Comment for later: we're throwing System.Exception in a number of places - we should either disable the rule or come up with a better solution.

File: src/SonarQube.Client/Services/SonarQubeClient.cs (hunk 1 above)
Comment (Evangelink) [reply src/SonarQube.Client/Services/SonarQubeClient.cs]: I will add more tests!

File: src/SonarQube.Client/Services/SonarQubeClient.cs (hunk 2 above)
Comment (Evangelink) [reply src/SonarQube.Client/Services/SonarQubeClient.cs]: I won't do anything for now but clearly my feeling is that we should keep the rule and create custom exceptions.

**Pull request closed without merging**